engine.py

```diff
@@ -196,7 +201,7 @@ def make_move(self, move):
 					master.destroy()
 					break
 
-	def undo_move(self, look_ahead_mode = False):
+	def undo_move(self, move, n, look_ahead_mode = False, undo=True):
```

Comment:
what is n?

Reply:
Is a parameter that allows undo_move to print only if u key is pressed

Reply:
Making your parameter names explicit helps with code readability. You could name it "allow_undo" for example